src/cmd/root.go

```diff
@@ -0,0 +1,230 @@
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			// This sets up loggers for all commands
+			err := setUpLoggers()
+			if err != nil {
+				return err
+			}
+
+			// Resolve the path to the toolbox binary
+			toolboxCmdPath, err := os.Executable()
+			if err != nil {
+				logrus.Fatalf("Failed to get absolute path to %s", os.Args[0])
+			}
+			viper.Set("TOOLBOX_CMD_PATH", toolboxCmdPath)
+			logrus.Debugf("Absolute path to %s is %s", os.Args[0], viper.Get("TOOLBOX_CMD_PATH"))
+
+			// Find out if the TOOLBOX_PATH env var is set
+			toolboxPath := viper.GetString("TOOLBOX_PATH")
+			calledCmd := cmd.CalledAs()
+			inContainer := utils.PathExists("/run/.containerenv")
+
+			if inContainer {
+				if toolboxPath == "" {
+					logrus.Fatal("TOOLBOX_PATH is not set")
+				}
+				if calledCmd != "init-container" {
+					logrus.Fatal("Toolbox currently does not work inside of a container. Please, run it on the host.")
+				}
+			} else {
+				if toolboxPath == "" {
+					viper.Set("TOOLBOX_PATH", viper.GetString("TOOLBOX_CMD_PATH"))
+				}
+			}
+			logrus.Debugf("TOOLBOX_PATH is %s", viper.GetString("TOOLBOX_PATH"))
+
+			// Set the toolbox runtime directory
+			viper.Set("TOOLBOX_RUNTIME_DIRECTORY", fmt.Sprintf("%s/toolbox", viper.GetString("XDG_RUNTIME_DIR")))
+			logrus.Debugf("Toolbox runtime directory is %s", viper.GetString("TOOLBOX_RUNTIME_DIRECTORY"))
+
+			// Check if it is needed to migrate to a new Podman version
+			// This doesn't have to be done in a container
+			if calledCmd != "init-container" && !inContainer {
+				err = migrate()
+				if err != nil {
+					logrus.Fatal(err)
+				}
+			}
+
+			// Here we could place some logic to take care of invoing toolbox or other commands from within container by piping them to the host
+			// TODO
+
+			return nil
+		},
+		Version:       version.GetVersion(),
+		SilenceErrors: true,
+	}
+)
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	rootCmd.PersistentFlags().BoolVarP(&rootFlags.assumeyes, "assumeyes", "y", false, "Automatically answer yes for all questions")
+	rootCmd.PersistentFlags().StringVar(&rootFlags.loglevel, "log-level", "warn", "Log messages above specified level: trace, debug, info, warn, error, fatal or panic")
+	rootCmd.PersistentFlags().BoolVar(&rootFlags.logPodman, "log-podman", false, "Show the log output of Podman. The log level is handled by the log-level option")
+	viper.BindPFlag("log-podman", rootCmd.PersistentFlags().Lookup("log-podman"))
+	// This flag is kept for compatibility reasons. In the future it would be better removed.
+	rootCmd.PersistentFlags().BoolVar(&rootFlags.verbose, "verbose", false, "Set log-level to 'debug'")
```

Comment:
We could turn `--verbose` into a *count* flag to trivially retain compatibility with `-vv`, abbreviated form of `--very-verbose`.